File: bugmailer/src/main/java/com/theah64/bugmailer/core/GitHubCommentGenerator.java
package com.theah64.bugmailer.core;

/**
 * Created by theapache64 on 8/2/18.
 */

public class GitHubCommentGenerator {
    private final StringBuilder builder = new StringBuilder();

    public GitHubCommentGenerator addNode(String key, String value) {
        builder.append("**").append(key).append("**").append("\n").append(value).append("\n\n");
        return this;
    }


    public GitHubCommentGenerator addNode(String key, int value) {
        return addNode(key, String.valueOf(value));
    }

    public GitHubCommentGenerator addCodeNode(String key, String value) {
        builder.append("**").append(key).append("**").append("\n```java\n").append(value).append("```\n\n");
        return this;
    }

    @Override
    public String toString() {
        return builder.toString();
    }
}

File: bugmailer/src/main/java/com/theah64/bugmailer/core/GitHubCommentGenerator.kt
package com.theah64.bugmailer.core

/**
 * Created by theapache64 on 8/2/18.
 */

class GitHubCommentGenerator {
    private val builder = StringBuilder()

    fun addNode(key: String, value: String): GitHubCommentGenerator {
        builder.append("**").append(key).append("**").append("\n").append(value).append("\n\n")
        return this
    }


    fun addNode(key: String, value: Int): GitHubCommentGenerator {
        return addNode(key, value.toString())
    }

    fun addCodeNode(key: String, value: String): GitHubCommentGenerator {
        builder.append("**").append(key).append("**").append("\n```java\n").append(value).append("```\n\n")
        return this
    }

    override fun toString(): String {
        return builder.toString()
    }
}
